### ISSUE TITLE
message timestamp test support

### ISSUE THREAD

**Issue by emmett9001** [Parsely/pykafka#692]
Tests around message timestamps should be added to `test_protocol.py::TestProduceAPI` and `test_protocol.py::TestFetchAPI`.

**Comment by emmett9001** [comment]
Found an issue with timestamp support during testwriting: the `timestamp` argument on `produce()` is ignored if `cluster._broker_version` is not set to `>=0.10`. `timestamp` being set on `produce()` should instead raise a loud error if the broker version does not support timestamps.

**Comment by emmett9001** [comment]
Currently I'm also seeing that messages produced with `protocol_version=1` are being returned from the broker to fetch requests with `protocol_version=0` when gzip compression is used. I'll have to dig more deeply into this.

**Comment by emmett9001** [comment]
https://github.com/Parsely/pykafka/pull/693 progress here

**Comment by emmett9001** [comment]
cc @maparent let me know if you have any insight into the compression issue I mentioned above. I see this behavior when using the pure python consumer.

**Comment by maparent** [comment]
Hmmm... You saw my last commit for compression; the compressed message has to have at least the protocol of the included messages. So it would be normal to get protocol-1 messages, even if not asking for them, if they contain a protocol 1 compressed payload. Could that be the scenario?

**Comment by maparent** [comment]
As for failing when you cannot use the timestamp, that's easy. I'll work on this later tonight if needed.

**Comment by emmett9001** [comment]
The situation is that after producing messages with protocol 1, the messages consumed from that topic (which match in every other respect) use protocol 0. I agree that I'd expect the logic you added to fix this problem, but that's not the behavior I'm seeing. Naively, it looks to me like the broker is not returning the correct protocol version specifier on messages in `FetchResponse`.

**Comment by maparent** [comment]
OK... had a first look and no great breakthrough, though I did fix a mistake (#694)
I have to concur with your assessment; the protocol version seems wrong on input. I tried bumping the FetchRequest's api_version but that blocks. I'll investigate more tomorrow.
I also did not set the protocol_version in the rd_kafka case. I could add it, but only by inferring from the presence of a timestamp; I delved in librdkafka, and that value is not passed on in the structs. It is not robust but the best I could do. Are you OK with that?
